src/app/InteractionModelEngine.cpp

```diff
@@ -193,83 +194,17 @@ void InteractionModelEngine::OnResponseTimeout(Messaging::ExchangeContext * ec)
     ChipLogProgress(DataManagement, "Time out! failed to receive echo response from Exchange: %d", ec->GetExchangeId());
 }
 
-InteractionModelEngine::HandlerKey::HandlerKey(chip::ClusterId aClusterId, chip::CommandId aCommandId,
-                                               Command::CommandRoleId aCommandRoleId) :
-    mClusterId(aClusterId),
-    mCommandId(aCommandId), mCommandRoleId(aCommandRoleId)
-{}
-
-inline bool InteractionModelEngine::HandlerKey::operator<(const HandlerKey & aOtherKey) const
-{
-    return ((mClusterId != aOtherKey.mClusterId)
-                ? (mClusterId < aOtherKey.mClusterId)
-                : ((mCommandId != aOtherKey.mCommandId) ? (mCommandId < aOtherKey.mCommandId)
-                                                        : ((mCommandRoleId < aOtherKey.mCommandRoleId))));
-}
-
-CHIP_ERROR InteractionModelEngine::RegisterClusterCommandHandler(chip::ClusterId aClusterId, chip::CommandId aCommandId,
-                                                                 Command::CommandRoleId aCommandRoleId, CommandCbFunct aDispatcher)
-{
-    CHIP_ERROR err = CHIP_NO_ERROR;
-
-    std::pair<HandlersMapType::iterator, bool> insertResult =
-        mHandlersMap.insert(HandlersMapType::value_type(HandlerKey(aClusterId, aCommandId, aCommandRoleId), aDispatcher));
-
-    if (!insertResult.second)
-    {
-        err = CHIP_ERROR_INVALID_ARGUMENT;
-    }
-    else
-    {
-        ChipLogDetail(DataManagement,
-                      "RegisterClusterCommandHandler registered handler for ClusterId = %d, CommandId = %d, CommandRoleId = "
-                      "%d, Dispatcher = %p",
-                      aClusterId, aCommandId, aCommandRoleId, (void *) aDispatcher);
-    }
-
-    return err;
-}
-
-CHIP_ERROR InteractionModelEngine::DeregisterClusterCommandHandler(chip::ClusterId aClusterId, chip::CommandId aCommandId,
-                                                                   Command::CommandRoleId aCommandRoleId)
+void __attribute__((weak))
```

Comment:
Why do we need this default implementation?

Reply:
Just to make linker happy before codegen for this is ready, added TODO to indicate that this should be removed.